### PR TITLE
OCPCLOUD-2594: Add periodic jobs as presubmit for CAS and CAO

### PR DESCRIPTION
**Pull request by racheljpg**
Hello! This is a PR to have our periodics from the [cluster-api-actuator-pkg](https://github.com/openshift/cluster-api-actuator-pkg) run as pre-submits in the [kubernetes-autoscaler](https://github.com/openshift/kubernetes-autoscaler) and [cluster-autoscaler-operator](https://github.com/openshift/cluster-autoscaler-operator). Thanks!

### CODE REVIEW

File: ci-operator/config/openshift/cluster-autoscaler-operator/openshift-cluster-autoscaler-operator-master.yaml [hunk 1]
```diff
@@ -98,6 +98,81 @@ tests:
         requests:
           cpu: 100m
     workflow: ipi-nutanix
+- always_run: true
+  as: e2e-aws-periodic-pre
```
Comment (JoelSpeed) [inline ci-operator/config/openshift/cluster-autoscaler-operator/openshift-cluster-autoscaler-operator-master.yaml:102]: For this repo, I would suggest having the AWS version as running always, but other platforms only on demand.

So for this one, just use `skip_if_only_changed` on AWS, drop `always_run`.

For the other platforms, drop the `skip_if_only_changed` and have `always_run: false`

File: ci-operator/config/openshift/kubernetes-autoscaler/openshift-kubernetes-autoscaler-master.yaml [hunk 2]
```diff
@@ -163,6 +163,81 @@ tests:
...
+      cli: latest
+      commands: JUNIT_DIR=${ARTIFACT_DIR} make test-e2e-periodic
+      from: src
+      resources:
+        requests:
+          cpu: 100m
+    workflow: ipi-aws
+- always_run: true
+  as: e2e-gcp-periodic-pre
+  skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
```
Comment (JoelSpeed) [inline ci-operator/config/openshift/kubernetes-autoscaler/openshift-kubernetes-autoscaler-master.yaml:183]: I don't think `always_run` and `skip_if_only_changed` are compatible, you should just have the skip and not the always run for this repo

**Comment by racheljpg** [comment]
pull-ci-openshift-kubernetes-autoscaler-master-e2e-aws-periodic-pre

**Comment by JoelSpeed** [comment]
Code changes look good here, would like to see a little more on the install on the rehearse before approving, it didn't even get to running the suite this time around

**Comment by racheljpg** [comment]
Hey @JoelSpeed, yep! I was having some trouble with the workflow needed for vsphere so just wanted to iterate a bit more rather than throwing them all up there together. But I'll start adding the others now (with your feedback above in mind :))

**Pull request merged**